molo/profiles/models.py

```diff
@@ -22,6 +22,12 @@ class UserProfilesSettings(BaseSetting):
         verbose_name=_('Mobile number required'),
     )
 
+    prevent_number_in_username = models.BooleanField(
+        default=False,
+        editable=True,
+        verbose_name=_('Prevent number in username / display name'),
```

Comment:
I would call this `prevent_phone_number_in_username` and change the `verbose_name` too so that it's clear what kind of number we're talking about here, ie: not just any number.